packages/sdk/src/addresses/index.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  ADDRESS_TYPE_TO_FORMAT,
+  NETWORK_TO_ADDRESS_TYPE_TO_REGEX,
+} from "./constants";
+import type { AddressFormat, AddressType } from "./types";
+import type { Network } from "../networks/types";
+
+export function getAddressFormat(
+  address: string,
+  network: Network,
+): AddressFormat {
+  const addressTypeToRegex = NETWORK_TO_ADDRESS_TYPE_TO_REGEX[network];
```

Comment:
Would it be better to use a proven library for this? Maybe https://www.npmjs.com/package/bitcoin-address-validation?